sdk/helper/custommetadata/custom_metadata.go

```diff
@@ -0,0 +1,66 @@
+package custommetadata
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-secure-stdlib/strutil"
+)
+
+type CustomMetadata map[string]string
```

Comment:
Last nit: as a consumer of this package, it's not immediately apparent to me what `CustomMetadata` is (other than a k/v mapping). Should we add some documentation here to describe the context in the PR comment?

Reply:
That's a great point, I'll add a comment.

Reply:
Nice!